catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java

```diff
@@ -914,65 +917,79 @@ CreateInfo parseAttachments(List<Attachment> contentParts, String transformerPar
+        metacardTypes.stream()
                 .map(metacardType -> metacardType.getAttributeDescriptor(parsedName))
                 .filter(Objects::nonNull)
                 .findFirst()
                 .map(AttributeDescriptor::getType)
-                .map(AttributeType::getAttributeFormat);
-        attributeFormat.ifPresent(attributeFormat1 -> {
-            try {
-                switch (attributeFormat1) {
-                case XML:
-                case GEOMETRY:
-                case STRING:
-                    attributes.add(new AttributeImpl(parsedName, IOUtils.toString(inputStream)));
-                    break;
-                case BOOLEAN:
-                    attributes.add(new AttributeImpl(parsedName, Boolean.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case SHORT:
-                    attributes.add(new AttributeImpl(parsedName, Short.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case LONG:
-                    attributes.add(new AttributeImpl(parsedName, Long.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case INTEGER:
-                    attributes.add(new AttributeImpl(parsedName, Integer.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case FLOAT:
-                    attributes.add(new AttributeImpl(parsedName, Float.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case DOUBLE:
-                    attributes.add(new AttributeImpl(parsedName, Double.valueOf(IOUtils.toString(
-                            inputStream))));
-                    break;
-                case DATE:
-                    Instant instant = Instant.parse(IOUtils.toString(inputStream));
-                    if (instant == null) {
-                        break;
-                    }
+                .map(AttributeType::getAttributeFormat)
+                .ifPresent((attributeFormat) -> parseAttribute(attributes,
+                        parsedName,
+                        inputStream,
+                        attributeFormat));
+    }
+
+    private void parseAttribute(List<Attribute> attributes, String parsedName,
+            InputStream inputStream, AttributeType.AttributeFormat attributeFormat) {
+        try (InputStream is = inputStream; InputStream boundedStream = new BoundedInputStream(is,
+                MAX_INPUT_SIZE + 1)) {
+            if (attributeFormat == OBJECT) {
+                LOGGER.debug("Object type not supported for override");
+                return;
+            }
+
+            byte[] bytes = IOUtils.toByteArray(boundedStream);
+            if (bytes.length > MAX_INPUT_SIZE) {
+                LOGGER.debug("Attribute length is limited to {} bytes", MAX_INPUT_SIZE);
```

Comment:
If I understand BoundedInputStream, it will never return more than MAX_INPUT_SIZE. It reads the wrapped input stream right up to MAX_INPUT_SIZE and then pretends like EOF was encountered. Should the RESTEndpoint fail instead of accepting the 65 KB it read? If the data is BINARY, then the 65 KB probably isn't useful. Maybe it should accept 65 KB of type String but cause a failure for other types?

Reply:
@ahoffer It will read up to` MAX_INPUT_SIZE + 1`. Meaning we only read one unnecessary byte and we are able to determine it was too large and treat it as a failure case. We do not process a partial/truncated value (which is why we have a `return` inside this if statement when `bytes.length > MAX_INPUT_SIZE`

Reply:
Missed that `return`. Thanks!